vm/systemSmartContracts/auction.go

```diff
@@ -1191,6 +1360,99 @@ func (s *stakingAuctionSC) unBondTokens(args *vmcommon.ContractCallInput) vmcomm
 	return vmcommon.Ok
 }
 
+func (s *stakingAuctionSC) unBondNecessaryNodes(registrationData *AuctionDataV2, valueToUnBond *big.Int) (*big.Int, error) {
+	totalUnBond := big.NewInt(0)
+	for _, blsKey := range registrationData.BlsPubKeys {
+		if !s.isWithStatusInactive(blsKey) {
+			continue
+		}
+
+		nodeData, errGet := s.getStakedData(blsKey)
+		if errGet != nil {
+			return nil, errGet
+		}
+		if len(nodeData.RewardAddress) == 0 {
+			continue
+		}
+		// returns what value is still under the selected bls key
+		vmOutput, errExec := s.executeOnStakingSC([]byte("unBond@" + hex.EncodeToString(blsKey)))
+		if errExec != nil || vmOutput.ReturnCode != vmcommon.Ok {
+			continue
+		}
+
+		registrationData.NumRegistered -= 1
+		totalUnBond.Add(totalUnBond, s.baseConfig.NodePrice)
+		totalUnBond.Sub(totalUnBond, nodeData.SlashValue)
```

Comment:
who pays for the slashValue? The unbonder will get his unbondValue and noone will pay for slashing until the last, that will not be able to unbond anything.

Reply:
that is part of the delegation contract which is not yet decided.